controllers/datadogagent/feature/remoteconfig/feature.go

```diff
@@ -53,16 +56,18 @@ func (f *rcFeature) ID() feature.IDType {
 func (f *rcFeature) Configure(dda *v2alpha1.DatadogAgent) (reqComp feature.RequiredComponents) {
 	f.owner = dda
 
-	if dda.Spec.Features != nil && dda.Spec.Features.RemoteConfiguration != nil && apiutils.BoolValue(dda.Spec.Features.RemoteConfiguration.Enabled) {
-		f.enabled = true
-		reqComp = feature.RequiredComponents{
-			Agent: feature.RequiredComponent{
-				IsRequired: apiutils.NewBoolPointer(true),
-				Containers: []apicommonv1.AgentContainerName{
-					apicommonv1.CoreAgentContainerName,
-				},
+	reqComp = feature.RequiredComponents{
+		Agent: feature.RequiredComponent{
+			IsRequired: apiutils.NewBoolPointer(true),
+			Containers: []apicommonv1.AgentContainerName{
+				apicommonv1.CoreAgentContainerName,
```

Comment:
are there conditions where the cluster agent is required to?

in general, i'm going to think about this pattern of having required components even when the feature is disabled..